jupyterhub/app.py

```diff
@@ -930,7 +930,7 @@ def _deprecate_api_tokens(self, change):
 
         with an :meth:`authenticate` method that:
 
-        - is a coroutine (asyncio or tornado)
+        - is a coroutine (asyncio)
```

Comment:
This is a major backward-incompatible change to no longer allow Spawners and Authenticators to use tornado `@gen.coroutine` or return any Future.

The biggest difference between tornado's implementation and asyncio's is that tornado supports strictly more cases. With tornado 6, at least, a tornado coroutine and an asyncio coroutine are the same thing, so this makes no difference, but implicit here is probably a new minimum tornado version. We should test this.

Reply:
Thanks for helping me understand this better, I have added back the comment for now.